operations/patch_util.go

```diff
@@ -212,44 +221,9 @@ func (p *patchParams) validatePatchCommand(ctx context.Context, conf *ClientSett
 		return
 	}
 
-	// update variants
-	if len(p.Variants) == 0 && p.Alias == "" {
-		p.Variants = conf.FindDefaultVariants(p.Project)
-		if len(p.Variants) == 0 && p.Finalize {
-			err = errors.Errorf("Need to specify at least one buildvariant with -v when finalizing." +
-				" Run with `-v all` to finalize against all variants.")
-			return
-		}
-	} else if p.Alias == "" {
-		defaultVariants := conf.FindDefaultVariants(p.Project)
-		if len(defaultVariants) == 0 && !p.SkipConfirm &&
-			confirm(fmt.Sprintf("Set %v as the default variants for project '%v'?",
-				p.Variants, p.Project), false) {
-			conf.SetDefaultVariants(p.Project, p.Variants...)
-			if err = conf.Write(""); err != nil {
-				grip.Warningf("warning - failed to set default variants: %v\n", err)
-			}
-		}
-	}
-
-	// update tasks
-	if len(p.Tasks) == 0 {
-		p.Tasks = conf.FindDefaultTasks(p.Project)
-		if len(p.Tasks) == 0 && p.Alias == "" && p.Finalize {
-			err = errors.Errorf("Need to specify at least one task or alias when finalizing." +
-				" Run with `-t all` to finalize against all tasks.")
-			return
-		}
-	} else if p.Alias == "" {
-		defaultTasks := conf.FindDefaultTasks(p.Project)
-		if len(defaultTasks) == 0 && !p.SkipConfirm &&
-			confirm(fmt.Sprintf("Set %v as the default tasks for project '%v'?",
-				p.Tasks, p.Project), false) {
-			conf.SetDefaultTasks(p.Project, p.Tasks...)
-			if err := conf.Write(""); err != nil {
-				grip.Warningf("warning - failed to set default tasks: %v\n", err)
-			}
-		}
+	if (len(p.Tasks) == 0 || len(p.Variants) == 0) && p.Alias == "" && p.Finalize {
+		err = errors.Errorf("Need to specify at least one task/variant or alias when finalizing.")
```

Comment:
I'd just return this on the next line.

Reply:
Isn't that what it's doing?

Reply:
I'd just combine these things on two lines